sql/analyzer/fix_exec_indexes.go

```diff
@@ -578,9 +580,23 @@ func (s *idxScope) visitSelf(n sql.Node) error {
 		}
 		if ne, ok := n.(sql.Expressioner); ok {
 			scope := append(s.parentScopes, s.childScopes...)
+			// default nodes can't see lateral join nodes, unless we're in lateral
+			// join and lateral scopes are promoted to parent status
 			for _, e := range ne.Expressions() {
-				// default nodes can't see lateral join nodes, unless we're in lateral
-				// join and lateral scopes are promoted to parent status
+				// groupConcat is special as it appends results to outer scope row
+				// we need to account for this extra column in the rows when assigning indexes
+				// see gms/expression/function/aggregation/group_concat.go:groupConcatBuffer.Update()
+				if gc, isGc := e.(*aggregation.GroupConcat); isGc {
```

Comment:
There are aggregate functions in Doltgres that need these semantics too, so we need a new marker interface for this instead of just a type check. Suggest sql.OrderedAggregate.